clang/lib/Parse/ParsePragma.cpp

```diff
@@ -1401,6 +1417,164 @@ bool Parser::HandlePragmaMSAllocText(StringRef PragmaName,
   return true;
 }
 
+NestedNameSpecifier *Parser::zOSParseIdentifier(StringRef PragmaName,
+                                                IdentifierInfo *IdentName) {
+  NestedNameSpecifier *NestedId = nullptr;
+  if (Tok.is(tok::coloncolon)) {
+    NestedId = NestedNameSpecifier::Create(Actions.Context, IdentName);
+  } else if (Actions.CurContext->isNamespace()) {
+    auto *NS = cast<NamespaceDecl>(Actions.CurContext);
+    NestedId =
+        NestedNameSpecifier::Create(Actions.Context, NS->getIdentifier());
+    NestedId =
+        NestedNameSpecifier::Create(Actions.Context, NestedId, IdentName);
+    PP.Lex(Tok);
+  } else {
+    NestedId = NestedNameSpecifier::Create(Actions.Context, IdentName);
+    PP.Lex(Tok);
+  }
+  while (Tok.is(tok::coloncolon)) {
+    PP.Lex(Tok);
+    IdentName = Tok.getIdentifierInfo();
```

Comment:
```suggestion
    IdentifierInfo *II = Tok.getIdentifierInfo();
```
Better to use a local variable instead of reusing the parameter.